src/app/design-model-module/service/design-model.service.ts

```diff
@@ -84,32 +110,94 @@ export class DesignModelService implements GraphDataService {
         const patternContainer = new PatternContainer();
         patternContainer.patterns = designModel._embedded ? designModel._embedded.patterns : [];
         patternContainer._links = designModel._links;
+        this.designModelLinks = patternContainer._links;
         return patternContainer;
       })
     );
   }
 
   getDesignModelByUrl(url: string): Observable<DesignModel> {
-    return this.http.get<DesignModel>(url);
+    return this.httpClient.get<DesignModel>(url);
   }
 
   getDesignModelByUuid(uuid: string): Observable<DesignModel> {
-    return this.http.get<DesignModel>(this.repoEndpoint + '/designModels/' + uuid + '/patterns');
+    return this.httpClient.get<DesignModel>(this.designModelsEndpoint + '/' + uuid + '/patterns');
+  }
+
+
+  getEdgeTypes(): Observable<string[]> {
+    if (!this.edgeTypes.getValue().length) {
+      this.httpClient.get<string[]>(this.designModelsEndpoint + '/edge-types').subscribe(
+        response => this.edgeTypes.next(response)
+      );
+    }
+
+    return this.edgeTypes.asObservable();
+  }
+
+
+  getEdges(): Observable<DirectedEdgeModel[] | UndirectedEdgeModel[]> {
+    return this.httpClient.get<any[]>(this.designModelLinks.edges.href)
+      .pipe(
+        map(edges => {
+          return edges.map(edge => {
+            edge.texts = [{
+              value: edge.type,
+              width: 100,
+              positionOnLine: 0.5,
+              offsetX: 8,
+              offsetY: -2,
+              clickEventKey: 'edge-type',
+              template: 'edge-label'
+            } as TextComponent];
+            return edge;
+          });
+        })
+      );
+  }
+
+
+  addEdge(edge: DirectedEdgeModel | UndirectedEdgeModel): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+      console.debug('Add edge', this.designModelLinks.edges.href, edge);
+      this.httpClient.post(this.designModelLinks.edges.href, edge).subscribe(
+        response => {
+          console.debug('Add edge response', response);
+          resolve();
+        }, error => {
+          console.error('Failed to add edge', error);
+          reject();
+        }
+      );
+    });
   }
 
+
+  deleteEdge(edge: { source: string, target: string }): void {
+    console.debug('Delete edge', this.designModelLinks.edges.href, edge);
+    this.httpClient.delete(this.designModelLinks.edges.href + '/' + edge.source + '/' + edge.target).subscribe(response => {
+      console.debug('Delete edge response', response);
```

Comment:
Should the user get feedback via a toast or something similar, rather than a debug message in the console?

Reply:
The deleted edge is removed from the graph, in my opinion this is enough user feedback. If required I can add a toast for additional feedback.